src/safir/testing/kubernetes.py

```diff
@@ -742,6 +861,435 @@ async def list_namespaced_event(
+        _request_timeout: Optional[int] = None,
+    ) -> V1JobList | Mock:
+        """List job objects in a namespace.
+
+        This does support watches.
+
+        Parameters
+        ----------
+        namespace
+            Namespace of jobs to list.
+        field_selector
+            Only ``metadata.name=...`` is supported. It is parsed to find the
+            job name and only jobs matching that name will be returned.
+        label_selector
+            Which events to retrieve when performing a watch.  All
+            labels must match.
+        resource_version
+            Where to start in the event stream when performing a watch. If
+            `None`, starts with the next change.
+        timeout_seconds
+            How long to return events for before exiting when performing a
+            watch.
+        watch
+            Whether to act as a watch.
+        _preload_content
+            Verified to be `False` when performing a watch.
+        _request_timeout
+            Ignored, accepted for compatibility with the watch API.
+
+        Returns
+        -------
+        kubernetes_asyncio.client.V1JobList or unittest.mock.Mock
+            List of jobs in that namespace, when not called as a watch. If
+            called as a watch, returns a mock ``aiohttp.Response`` with a
+            ``readline`` metehod that yields the events.
+
+        Raises
+        ------
+        AssertionError
+            Some other ``field_selector`` was provided.
+        kubernetes_asyncio.client.ApiException
+            Raised with 404 status if the namespace does not exist.
+        """
+        self._maybe_error("list_namespaced_job", namespace, field_selector)
+        if namespace not in self._objects:
+            msg = f"Namespace {namespace} not found"
+            raise ApiException(status=404, reason=msg)
+        label_dict = {}
+        if label_selector:
+            label_dict = _parse_label_selector(label_selector)
+        if not watch:
+            if field_selector:
+                match = re.match(r"metadata\.name=(.*)$", field_selector)
+                assert match and match.group(1)
+                try:
+                    job = self._get_object(namespace, "Job", match.group(1))
+                    return V1JobList(kind="Job", items=[job])
+                except ApiException:
+                    return V1JobList(kind="Job", items=[])
+            else:
+                jobs = []
+                if "Job" in self._objects[namespace]:
+                    for obj in self._objects[namespace]["Job"].values():
+                        if label_dict:
+                            job_labels = obj.metadata.labels
+                            for lbl in label_dict:
+                                if (
+                                    lbl not in job_labels
+                                    or label_dict[lbl] != job_labels[lbl]
+                                ):
+                                    continue
```

Comment:
This still has that bug where the for loop doesn't do anything because the continue is within the loop. I think you want to call your new helper function.

Reply:
Yeah, missed this one too.